pkg/ccl/backupccl/backup_job.go

```diff
@@ -664,18 +681,229 @@ func (b *backupResumer) readManifestOnResume(
+		// No reason to return an error which might cause problems if it doesn't
+		// seem to exist.
+		log.Warningf(ctx, "failed to release protected which seems not to exist: %v", err)
+		return nil
+	}
+	return err
+}
+
+// manageFullBackupPTSChaining is invoked on successful completion of a
+// scheduled full backup. It is responsible for:
+// - Releasing the pts record that was stored on the incremental schedule when
+//   the full backup was planned.
+// - Writing a new pts record protecting all data after the full backups' EndTime
+//   and store this on the incremental schedule.
+func manageFullBackupPTSChaining(
+	ctx context.Context,
+	env scheduledjobs.JobSchedulerEnv,
+	txn *kv.Txn,
+	backupDetails jobspb.BackupDetails,
+	exec *sql.ExecutorConfig,
+	args *ScheduledBackupExecutionArgs,
+) error {
+	// Let's resolve the dependent incremental schedule as the first step. If the
+	// schedule has been dropped then we can avoid doing unnecessary work.
+	incSj, _, err := getScheduledBackupExecutionArgsFromSchedule(ctx, env, txn, exec.InternalExecutor,
+		args.DependentScheduleID)
+	if err != nil {
+		if jobs.HasScheduledJobNotFoundError(err) {
+			log.Warningf(ctx, "could not find dependent schedule with id %d",
+				args.DependentScheduleID)
+			return nil
+		}
+		return err
+	}
+
+	// Resolve the spans that need to be protected on this execution of the
+	// scheduled backup.
+	spansToProtect, err := getSpansProtectedByBackup(ctx, backupDetails, txn, exec)
+	if err != nil {
+		return errors.Wrap(err, "getting spans to protect")
+	}
+
+	// Protect the spans after the EndTime of the current backup. We do not need
+	// to verify this new record as we have a record written by the backup during
+	// planning, already protecting these spans after EndTime.
+	//
+	// Since this record will be stored on the incremental schedule, we use the
+	// inc schedule ID as the records' Meta. This ensures that even if the full
+	// schedule is dropped, the reconciliation job will not release the pts
+	// record stored on the inc schedule, and the chaining will continue.
+	ptsRecord, err := protectTimestampRecordForSchedule(ctx, spansToProtect,
+		backupDetails.EndTime, incSj.ScheduleID(), exec, txn)
+	if err != nil {
+		return errors.Wrap(err, "protect and verify pts record for schedule")
+	}
+
+	// Attempt to release the pts record that was written on the incremental
+	// schedule when the full backup was being planned.
+	if err := releaseProtectedTimestampRecordForSchedule(ctx, txn,
+		backupDetails.ReleaseOnSuccessPTSRecord, exec); err != nil {
+		return errors.Wrap(err, "release pts record for schedule")
+	}
+
+	// Update the incremental schedule with the new pts record.
+	incSj.ExecutionArgs().ProtectedTimestampRecord = &ptsRecord
+	incSj.SetExecutionDetails(
+		incSj.ExecutorType(),
+		*incSj.ExecutionArgs(),
+	)
+	return incSj.Update(ctx, exec.InternalExecutor, txn)
+}
+
+func (b *backupResumer) maybeUpdateProtectedTimestampRecordOnSchedule(
```

Comment:
Nit: `OnSchedule` sounds like it's triggered by the schedule. Perhaps `maybeUpdateSchedulePTSRecord`?

Reply:
done.